packages/react-core/src/components/TreeView/TreeViewListItem.tsx

```diff
@@ -94,14 +97,23 @@ const TreeViewListItemBase: React.FunctionComponent<TreeViewListItemProps> = ({
   }, [isExpanded, defaultExpanded]);
 
   const Component = hasCheck ? 'div' : 'button';
-  const ToggleComponent = hasCheck ? 'button' : 'div';
+  let ToggleComponent: 'div' | 'span' | 'button' = 'div';
+  if (hasSelectableNodes) {
+    ToggleComponent = 'span';
+  }
+  if (hasCheck) {
+    ToggleComponent = 'button';
+  }
```

Comment:
...and having the toggle set to a button when either of these props are passed in.

Plain keyboard functionality would still require some tweaking to get working properly, though.

Reply:
I had both of these component types set up the way you are describing, but the structure didn't match core so I backed them out. It looks like there's a few more updates that this PR needs to fully match core (below) so hopefully that will iron out the behavior a bit.

Reply:
@kmcfaul updated the core structure so that selectable/expandable nodes are now a `div` with a `tabindex` so you can focus them, and the toggle inside is now a `button`. https://github.com/patternfly/patternfly/pull/4991

You should be able to make that change in your PR without pulling in core, there is no CSS or anything needed to support that change.